### PR TITLE
app-layer-pop3: Add POP3 support.

### PR DESCRIPTION
**Pull request by decanio**
Add POP3 application layer parser and eve-log JSON logger.

https://buildbot.openinfosecfoundation.org/builders/decanio/builds/29
https://buildbot.openinfosecfoundation.org/builders/decanio-pcap/builds/29

### CODE REVIEW

File: src/app-layer-pop3.c [hunk 1]
```diff
@@ -0,0 +1,1080 @@
...
+            }
+        }
+        files = pop3_state->files_ts;
+
+        /* Open file if necessary */
+        if (state->body_begin) {
+
+            if (SCLogDebugEnabled()) {
+                SCLogDebug("Opening file...%u bytes", len);
+                printf("File - ");
```
Comment (gozzy) [inline src/app-layer-pop3.c]: These printfs look like a debug output.

Comment (decanio) [inline src/app-layer-pop3.c]: Yes.  This was put in during initial development.  Its served its purpose.  I'll pull it out.

**Comment by decanio** [comment]
I'm happy to add support for AFL to this. Do you want this as a part of this PR or a separate one that gets applied on top of this one? Remember I'm already planning on sending you additional PRs that add "filedata" and LUA support to pop3.

**Comment by inliniac** [comment]
I'd like to run the AFL test before merging anything, so as part of the PR would be great :)

**Pull request closed without merging**